plugins/SegmentEditor/tests/System/ApiTest.php

```diff
@@ -62,4 +62,50 @@ public function test_segmentHashWorkflow_whenSegmentIsCrazyEncoded()
 
         $this->assertEquals($segmentApiHash, $segmentDefinitionHash);
     }
-}
\ No newline at end of file
+
+    /**
+    * @dataProvider segmentsDataProvider
+    */
+    public function test_generatedSegmentHash($segment)
+    {
+        Fixture::createWebsite('2020-03-03 00:00:00');
+
+        Config::getInstance()->General['enable_browser_archiving_triggering'] = 0;
+        self::$fixture->getTestEnvironment()->overrideConfig('General', 'enable_browser_archiving_triggering', 0);
+        self::$fixture->getTestEnvironment()->save();
+
+        $url = Fixture::getTestRootUrl() . '?' . http_build_query([
+            'module' => 'API',
+            'method' => 'SegmentEditor.add',
+            'name' => 'test segment',
+            'definition' => $segment,
+            'idSite' => 1,
+            'autoArchive' => 1,
+            'enabledAllUsers' => 1,
+            'format' => 'json',
+            'token_auth' => Fixture::getTokenAuth(),
+        ]);
+        self::assertStringContainsString(urlencode($segment), $url);
+
+        Http::sendHttpRequest($url, 10);
+
+        $segments = SegmentEditorApi::getInstance()->getAll();
```

Comment:
btw here you will want to fetch a specific segment using the `get($idSegment)` method and not all of them maybe. When you use above `$idSegment = SegmentEditorApi::getInstance()->add()` then you will also have the ID.

Reply:
just in case the segments returned... the new one might not be at the end (unless there is some sort)